### PR TITLE
Updated module with 2.5+ requirements and example

### PR DESCRIPTION
**Pull request by mglantz**
I've updated the documentation with 2.5+ requirements plus a working example of implementing them.

load_balancing_rules requires frontend_port and also requires backend_port to be passed if not the user has taken special action to enable additional features on the azure subscription. I marked both as required as the backend_port error message doesn't really explain that backend_port needs to be defined except for in a special scenario.

Error message as follows
'''
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error creating or updating load balancer myLoadBalancer - Azure Error: SubscriptionNotRegisteredForFeature\nMessage: Subscription asdfasdfasdfasdfasdfasdfasdfasdfasdf is not registered for feature Microsoft.Network/AllowILBAllPortsRule required to carry out the requested operation."}
'''

+label: docsite_pr

##### ISSUE TYPE

 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by yuwzho** [comment]
shipit

Unfortunately your test together with other tests run out of the subscription's resource limit. Please close this pr and reopen it to trigger another build.

**Comment by yuwzho** [comment]
bot_status

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 1]
```diff
@@ -189,11 +189,13 @@
                     - Frontend port numbers must be unique across all rules within the load balancer.
                     - Acceptable values are between 0 and 65534.
                     - Note that value 0 enables "Any Port"
+                required: True
             backend_port:
                 description:
                     - The port used for internal connections on the endpoint.
                     - Acceptable values are between 0 and 65535.
                     - Note that value 0 enables "Any Port"
+                required: True
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: backend port is not required

Comment (mglantz) [reply lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: Hello there, yes, that is true. Please see first post for an argument to set it True. The feature to be activated on the Azure account is not activated by default and the error message is close to garbage for most people. By default it's required, it's not required for people who has activated AllowILBAllPortsRule. IMHO it then makes sense to have it required.

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: Well, the documentation should align the spec. In the spec the backend_port in inat is required but not in lb rules

https://github.com/ansible/ansible/blob/00884581ffabed74ce9ea3ee9f99a7d34cfffd56/lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py#L461-L463

**Review by yuwzho — CHANGES_REQUESTED** [review]

**Comment by yungezz** [comment]
@mglantz  could you pls resolve conflicts?

**Comment by Fred-sun** [comment]
@mglantz Thanks for you contribution,  Could you pls resolve conflicts?

**Comment by mglantz** [comment]
Hi all, now back from vacation... :) I removed the requirement for backend port and resolved conflict.

**Comment by mglantz** [comment]
@Fred-sun is this better?

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
@mglantz It looks ok from my side, but the bot will prevent the merge since there is a merge commit. You may need to rebase your code to devel branch and force push again. Then trigger CI again.

**Comment by mglantz** [comment]
@yuwzho I've rebased on devel, seems there's a test failing though.

**Comment by Fred-sun** [comment]
@yuwzho @yungezz Could you help take a look this PR? Thanks!

**Comment by yuwzho** [comment]
@mglantz Seems this is not related to your changes, but also the lb's documentation sanity test failed. From line 352, there need one more indent. Would you mind changing these line also?

**Comment by yuwzho** [comment]
from line 310 - line 351, are over indent

https://github.com/ansible/ansible/blob/0eeee742870b66542b14e904d14fb2776f739976/lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py#L310-L351

**Comment by Fred-sun** [comment]
@mglantz Thanks for the contribution, Could you help to recheck the PR according by @yuwzho's advice? Thanks !

**Comment by Fred-sun** [comment]
@mglantz Thanks for the contribution, Could you help to recheck the PR according by @yuwzho's comment? Thanks !

**Comment by Fred-sun** [comment]
@mglantz Thanks for the contribution, It need you change according by @yuwzho's advice. Thanks!

**Comment by Fred-sun** [comment]
@mglantz

**Comment by Fred-sun** [comment]
@mglantz Thanks for the contribution, Are you still here? Could you help change according by @yuwzho's comment? Thanks!

**Comment by mglantz** [comment]
Hey @Fred-sun, sorry for this taking 1000 years. I've fixed it.

**Comment by mglantz** [comment]
@Fred-sun I can't see any errors coming out of Shippable, even though it seems to say that all checks have failed. Let me know. I don't have anything competing for my attention except for this now.

**Comment by Fred-sun** [comment]
@mglantz Thanks for you update, As @yuwzho describe, From line 310 - line 351 both need to be indented two space forward. Thanks!

**Comment by mglantz** [comment]
@Fred-sun Forward, aah.

**Comment by Fred-sun** [comment]
@yuwzho Could you help to review this PR again? The contributor has changed. Thanks!

**Comment by mglantz** [comment]
> @yuwzho Could you help to review this PR again? The contributor has changed. Thanks!

I've changed it so that I follow the same indention as in https://github.com/mglantz/ansible/blob/patch-1/lib/ansible/modules/cloud/azure/azure_rm_aks.py

**Comment by mglantz** [comment]
> The test `ansible-test sanity --test validate-modules` [[explain](https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/validate-modules.html)] failed with 1 error:
> 
> ```
> lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:323:7: E311 EXAMPLES is not valid YAML
> ```
> The test `ansible-test sanity --test yamllint` [[explain](https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/yamllint.html)] failed with 1 error:
> 
> ```
> lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:323:7: error EXAMPLES: syntax error: expected <block end>, but found '-'
> ```
> [click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

Fixing..

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 2]
```diff
@@ -306,32 +307,55 @@
...
-        backend_address_pool: backendaddrpool0
-        frontend_port: 80
-        backend_port: 80
-        probe: prob0
+    # Example for 2.5+ module
+    - name: Create a master load balancer
+      azure_rm_loadbalancer:
+        name: myLoadBalancer
+        resource_group: myResourceGroup
+        location: northeurope
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:315]: location is not necessary

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 3]
```diff
@@ -306,32 +307,55 @@
...
+        load_balancing_rules:
+          - name: myLoadBalancingRules
+            frontend_port: 8443
+            backend_port: 8443
+            frontend_ip_configuration: myFrontendIPcfg
+            load_distribution: SourceIP
+            protocol: Tcp
+            backend_address_pool: myBackendPool
+            probe: myProbe
+      # Example for <2.5 module
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:337]: i dont think this is necessary to keep old one. docs has version also

**Comment by Fred-sun** [comment]
@mglantz Thanks for your contribution, Could you help update according by yungezz's comment? Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@mglantz Thanks for the contrition, Are you still here? Could you help give a change according by yungezz and yuwzho's comment? Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
kindly ping

**Comment by yungezz** [comment]
pls rebase and resolve comments.

**Comment by Fred-sun** [comment]
@mglantz Thanks for your contribution. Could you help rebase the PR and resolve the comments? Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@mglantz  Are you still here? Could you help rebase the PR and resolve the comments? Thanks!

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py (hunk 2 above)
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py:315]:
```suggestion
```

**Review by acozine — CHANGES_REQUESTED** [review]
Thanks @mglantz for taking the time to update the Ansible documentation.

I agree with @yungezz that we should remove the example for pre-2.5 - those examples will remain in the 2.5 docs. If you do that, rebase, and ping me back, I'll try to merge this quickly, before other changes go in on the same module.

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@mglantz Are you still here? Please help resolve the conflicting files and change according by above comments! Thanks!

**Comment by Fred-sun** [comment]
@yungezz The PR has no update more than five month, and exists conflicting file, Could we can close this? also can create a new to instead of it when this function need! Thanks!

**Comment by yungezz** [comment]
close it for now, due to conflict and no response on comments.  ansible docs has version,  so devel branch is always latest sample. if user need check sample of old version, can check doc  of old version.  pls reopen it if necessary we can continue the discussion.

**Pull request closed without merging**